crates/noirc_evaluator/src/ssa_refactor/ssa_builder/mod.rs

```diff
@@ -226,11 +246,14 @@ impl FunctionBuilder {
 
     /// Insert a constrain instruction at the end of the current block.
     pub(crate) fn insert_constrain(&mut self, boolean: ValueId) {
-        self.insert_instruction(Instruction::Constrain(boolean), None);
+        let results = self.insert_instruction(Instruction::Constrain(boolean), None).results();
+        assert!(results.is_empty(), "constrain instructions do not return any results");
+        // TODO: maybe put as a method on results like `.first`
```

Comment:
I don't think this assert is necessary since we control the `insert_instruction` method and the constrain case isn't complex enough to warrant asserts that we're doing it correctly. It is something that is unlikely to change in the future as well. I generally think asserts are reasonable across abstraction boundaries but I consider this to be an implementation detail within the SSA IR.